src/auth/OIDCAuth.test.ts

```diff
@@ -0,0 +1,145 @@
+import type { Mock, MockInstance } from "vitest";
+
+import * as jimmListeners from "juju/jimm/listeners";
+import * as jimmThunks from "juju/jimm/thunks";
+import { thunks as appThunks } from "store/app";
+import { actions as generalActions } from "store/general";
+
+import { Auth } from "./Auth";
+import { OIDCAuth } from "./OIDCAuth";
+
+describe("OIDCAuth", () => {
+  let dispatch: Mock;
+
+  beforeEach(() => {
+    // @ts-expect-error - Resetting singleton for each test run.
+    delete Auth.instance;
+
+    dispatch = vi.fn();
+    new OIDCAuth(dispatch);
+  });
+
+  afterEach(() => {
+    vi.restoreAllMocks();
+  });
+
+  describe("boostrap", () => {
+    let addWhoamiListenerMock: MockInstance;
+    let connectAndStartPollingMock: MockInstance;
+
+    beforeEach(() => {
+      addWhoamiListenerMock = vi.spyOn(jimmListeners, "addWhoamiListener");
+      connectAndStartPollingMock = vi.spyOn(
+        appThunks,
+        "connectAndStartPolling",
+      );
+    });
+
+    it("connection success", async () => {
+      dispatch.mockResolvedValue({ payload: {} });
+
+      await Auth.instance.bootstrap();
+
```

Comment:
A tiny note, but there's a bit of unnecessary whitespace in some of these tests.